DOCUMENTATION.md

```diff
@@ -658,6 +658,51 @@ If the request is not successful, the client returns a `Notifications::Client::R
 |`403`|`AuthError: Error: Your system clock must be accurate to within 30 seconds`|`AuthError`|Check your system clock|
 |`403`|`AuthError: Invalid token: signature, api token not found`|`AuthError`|Use the correct API key. Refer to [API keys](/ruby.html#api-keys) for more information|
 
+## Get a PDF for a letter notification
+
+### Method
+
+This returns the pdf contents of a letter notification.
+
+```ruby
+pdf_file = client.get_pdf_for_letter(
+  'f33517ff-2a88-4f6e-b855-c550268ce08a' # notification id (required)
+)
+```
+
+### Arguments
+
+#### id (required)
+
+The ID of the notification. You can find the notification ID in the response to the [original notification method call](/python.html#get-the-status-of-one-message-response).
+
+You can also find it in your [GOV.UK Notify Dashboard](https://www.notifications.service.gov.uk).
+
+1. Sign into GOV.UK Notify and select __Dashboard__.
+1. Select __letters sent__.
+1. Select the relevant notification.
+1. Copy the notification ID from the end of the page URL, for example `https://www.notifications.service.gov.uk/services/af90d4cb-ae88-4a7c-a197-5c30c7db423b/notification/ID`.
+
+### Response
+
+If the request to the client is successful, the client will return a `string` containing the raw PDF data.
+
+### Error codes
+
+If the request is not successful, the client throws a `Notifications::Client::RequestError` and an error code.
+
+|error.status_code|error.message|How to fix|
```

Comment:
The error tables for this client have an extra column called `class` (because we've added lots more error classes in `request_error.rb`). So would probably be good to add that column here to keep things consistent.

Reply:
hmmm - thinking of client errors, do you reckon it's worth spinning out a new exception `PdfNotReadyError`? Since how you handle a PdfNotReady is pretty different to other errors (wait a few seconds then try again vs changing something about how you made the call with other 400s)